content/documentation/pages/3-stream-developer-guides/2-streams/1-standalone-stream-sample.md

```diff
@@ -0,0 +1,641 @@
+
+# Building the downloaded sample
+
+The stream apps may may be configured to run with a Kafka broker or RabbitMQ, using a common code base. The only difference is in the executable jar files. In order for them to work with a Kafka broker, they require a Kafka binder dependency (enabled by default). For RabbitMQ, they require the Rabbit binder.
+
+<!--TIP-->
+
+Additional Spring Cloud Stream binder implementations include Kafka Streams, Amazon Kinesis, Google PubSub (partner maintained), Solace PubSub+ (partner maintained), Azure Event Hubs (partner maintained). The binder selection is a build time step. The sample project uses a Maven profile to enable the corresponding binder.
+
+<!--END_TIP-->
+
+<!--TABS-->
+
+<!--Kafka-->
+
+To build the sample stream app for Kafka, from the root project directory:
+
+```bash
+$./mvnw clean package -Pkafka
+```
+
+<!--RabbitMQ-->
+
+To build the sample stream app for RabbitMQ, from the root project directory:
+
+```bash
+$./mvnw clean package -Prabbit
+```
+
+<!--END_TABS-->
+
+# Development
+
+We create three Spring Cloud Stream applications that communicate by using the configured binder.
+
+The scenario is a cell phone company creating bills for its customers.
+Each call made by a user has a `duration` and an amount of `data` used during the call.
+As part of the process to generate a bill, the raw call data needs to be converted to a cost for the duration of the call and a cost for the amount of data used.
+
+The call is modeled by using the `UsageDetail` class that contains the `duration` of the call and the amount of `data` used during the call.
+The bill is modeled by using the `UsageCostDetail` class that contains the cost of the call (`costCall`) and the cost of the data (`costData`). Each class contains an ID (`userId`) to identify the person making the call.
+
+The three streaming applications are as follows:
+
+- The `Source` application (named `UsageDetailSender`) generates the users' call `duration` and the amount of `data` used for each `userId` and sends a message that contains the `UsageDetail` object as JSON.
+
+- The `Processor` application (named `UsageCostProcessor`) consumes the `UsageDetail` and computes the cost of the call and the cost of the data per `userId`. It sends the `UsageCostDetail` object as JSON.
+
+- The `Sink` application (named `UsageCostLogger`) consumes the `UsageCostDetail` object and logs the cost of the call and data.
+
+## Source
+
+In this step, we create the `UsageDetailSender` source.
+
+<!--TABS-->
+
+<!--Kafka-->
+
+You can either [directly download the project generated by the Spring Initialzr](https://start.spring.io/#!type=maven-project&language=java&platformVersion=2.4.3.RELEASE&packaging=jar&jvmVersion=1.8&groupId=io.spring.dataflow.sample&artifactId=usage-detail-sender-kafka&name=usage-detail-sender-kafka&description=Demo%20project%20for%20Spring%20Boot&packageName=io.spring.dataflow.sample.usagedetailsender&dependencies=cloud-stream,actuator,kafka)
```

Comment:
Takes you to start.spring.io with the site properly populated, but you have to click the generate button to download the page.